app/RemoteSite/Connection.php

```diff
@@ -121,12 +122,33 @@ protected function performHttpRequest(
         /** @var Client $httpClient */
         $httpClient = App::make(Client::class);
 
+        // Send a streamed response to be able to validate the size
+        $options['stream'] = true;
+        $options['progress'] = function (
+            $downloadTotal,
+            $downloadedBytes
+        ) use ($request) {
+            if ($downloadedBytes > 1024000) {
+                throw new \RuntimeException("Unplausible response size while fetchting from " . $request->getUri());
```

Comment:
```suggestion
                throw new \RuntimeException("Unplausible response size while fetching from " . $request->getUri());
```

correct spelling of fetching
unplausible is not english as well but i dont know what you meant